reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt

```diff
@@ -90,19 +94,21 @@ class AntennaAttributionDocumentReporter : Reporter {
                     "The template path does not point to a template file."
                 }
 
-                templateId = id
-                addTemplateToClasspath(templatePath.toURI().toURL())
+                addTemplateToClassPath(templatePath.toURI().toURL()).also {
+                    templateId = id
+                }
             }
         }
 
-        val workingDir = createTempDir()
         val documentFile = AttributionDocumentGeneratorImpl(
             defaultFilename,
             workingDir,
             templateId,
             DocumentValues(rootProject.id.name, rootProject.id.version, projectCopyright)
         ).generate(artifacts)
 
+        if (templateId != DEFAULT_TEMPLATE_ID) removeTemplateFromClassPath()
```

Comment:
Shouldn't this be in a `finally` block to not risk leaving the Thread with the wrong classloader?

Reply:
Yeah, that's a good idea, done. Although keeping the class loader probably wouldn't have done any harm as it's a child class loader of the original class loader.